lib/tfpdf/wp-tfpdf.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * To avoid modifying tFPDF directly we introduce WP_tFPDF, 
+ * A thin wrapper around the parent class to utilize WP_Filesystem 
+ * 
+ * Parent's coding standards are preserved deliberately.
+ */
+// phpcs:disable WordPress.WP.AlternativeFunctions.file_system_read_fwrite, WordPress.WP.AlternativeFunctions.file_system_read_fread, WordPress.WP.AlternativeFunctions.file_system_read_fopen, WordPress.WP.AlternativeFunctions.file_system_read_fclose, WordPress.VIP.FileSystemWritesDisallow
+if ( ! defined( 'ABSPATH' ) ) exit; // Exit if accessed directly
+
+require_once ABSPATH . '/wp-admin/includes/file.php';
+
+class WP_tFPDF extends tFPDF {
+	function __construct( $orientation = 'P', $unit = 'mm', $size = 'A4' ) {
+		parent::__construct( $orientation, $unit, $size );
+		$this->init_wp_filesystem();
+	}
+	
+	function init_wp_filesystem() {
+		global $wp_filesystem;
+
+		if ( ! is_a( $wp_filesystem, 'WP_Filesystem_Base' ) ) {
+			$creds = request_filesystem_credentials( site_url() );
+			$fs = wp_filesystem( $creds );
+			return $fs ?: new WP_Error( 'fs-init-error', "Couldn't initialize Filesystem" );
+		}
+		
+		return true;
+	}
+	
+	function Output($name='', $dest='')
+{	
+	// Output PDF to some destination
+	if($this->state<3)
+		$this->Close();
+	$dest = strtoupper($dest);
+	if($dest=='')
+	{
+		if($name=='')
+		{
+			$name = 'doc.pdf';
+			$dest = 'I';
+		}
+		else
+			$dest = 'F';
+	}
+	switch($dest)
+	{
+		case 'I':
+			// Send to standard output
+			$this->_checkoutput();
+			if(PHP_SAPI!='cli')
+			{
+				// We send to a browser
+				header('Content-Type: application/pdf');
+				header('Content-Disposition: inline; filename="'.$name.'"');
+				header('Cache-Control: private, max-age=0, must-revalidate');
+				header('Pragma: public');
+			}
+			echo $this->buffer;
+			break;
+		case 'D':
+			// Download file
+			$this->_checkoutput();
+			header('Content-Type: application/x-download');
+			header('Content-Disposition: attachment; filename="'.$name.'"');
+			header('Cache-Control: private, max-age=0, must-revalidate');
+			header('Pragma: public');
+			echo $this->buffer;
+			break;
+		case 'F':
```

Comment:
Correct, there's no obvious code path to get the file to be written. The one I used is actually calling the method inside a `sensei_certificates_before_pdf_output` action hook. 

In all honesty, this can probably be removed without anyone noticing.  I just didn't want to trample on the flowers in someone else's garden :)